odoo/models.py

```diff
@@ -5039,7 +5041,7 @@ def _revert_method(cls, name):
     #  - the global cache is only an index to "resolve" a record 'id'.
     #
 
-    def __init__(self, env, ids, prefetch_ids):
+    def __init__(self, env, ids=(), prefetch_ids=()):
```

Comment:
We could even consider replacing `browse` entirely with the class trick:
```py
from odoo.addons.base.models import ResPartner, ResUsers

class Foo(Model):
    def meth(self):
        superuser = ResUsers(self.env, 1)           # same as self.env['res.users'].browse(1)
        partners = ResPartner(self.env).search([])  # same as self.env['res.partner'].search([])
```
Going one step further, we could put `env` in a context var, and make `self.env` almost _invisible_ :wink:

Reply:
@rco-odoo As suggested, I modified a bit the code of our module patching the Odoo metaclass to allow to define an `_inherit` of an Odoo model by simple python inheritance (no need for the 'extends=True' argument) see https://github.com/sbidoul/typodoo/pull/2/files

Then, you can still write this, as usual:

```python
from odoo import models, fields

class ResPartner(models.Model):
    _inherit = "res.partner"

    friend = fields.Boolean()
    industry2_id = fields.Many2one(
        'res.partner.industry',
        string='Secondary Industry',
        required=True,
    )
```

But you can also write this:

```python
from odoo import fields

from odoo.addons.base.models.res_partner import (
    Partner, ResPartnerIndustry
)

class ResPartner(Partner):
    friend: bool = fields.Boolean()
    industry2_id = fields.Many2one[ResPartnerIndustry](
        string='Secondary Industry',
        required=True,
    )
```